static/js/recent_topics_ui.js

```diff
@@ -90,6 +90,12 @@ const ls_key = "recent_topic_filters";
 const ls = localstorage();
 
 let filters = new Set();
+if (!page_params.is_spectator) {
+    // A user may have a stored filter and can log out
+    // to see web public view. This ensures no filters are
+    // selected for spectators.
+    filters = new Set(ls.get(ls_key));
+}
```

Comment:
Let's do this in an `initialize` method -- we aim to never do anything at import time, since it's bad for testing among other things.